server/src/main/java/com/unosquare/carmigo/controller/JourneyController.java

```diff
@@ -46,11 +49,20 @@ public ResponseEntity<JourneyDriverViewModel> getJourneyById(@PathVariable final
         return ResponseEntity.ok(journeyDriverViewModel);
     }
 
-    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    @GetMapping(value = "/search", produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.OK)
-    public ResponseEntity<List<JourneyDriverViewModel>> getJourneys()
+    public ResponseEntity<List<JourneyDriverViewModel>> searchJourneys(
+            @RequestParam("locationIdFrom") final String locationIdFrom,
+            @RequestParam("locationIdTo") final String locationIdTo,
+            @RequestParam("dateTimeFrom") final String dateTimeFrom,
+            @RequestParam("dateTimeTo") final String dateTimeTo)
     {
-        final List<GrabJourneyDTO> grabJourneyDTOList = journeyService.getJourneys();
+        final Map<String, String> params = new HashMap<>();
```

Comment:
Hey Kleyton.  Spring Boot allows for implicit mapping from an endpoints request parameters to a map.  e.g

```
@RequestMapping(value = {"/search/", "/search"}, method = RequestMethod.GET)
public String search(
@RequestParam Map<String,String> allRequestParams, ModelMap model) {
   return "viewName";
}
```

```
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ProductByItemNumberAndGroupCriteria {

  @NotNull
  @NotEmpty
  private String itemNumber;

  @NotNull
  private int groupId;
  @NotNull
  private int groupOptionId;

}
```



Or, you can use what is termed as criteria.  This involves creating a class which will hold the parameters.  Spring will automatically do this for you.  You can also add validation to the fields.  example:

 ```
@GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
  @ApiOperation(value = "Retrieves a product based on a newly selected group option")
  public ResponseEntity<ProductSummaryViewModel> getProductByGroupOption(final ProductByItemNumberAndGroupCriteria productByItemNumberAndGroupCriteria) {
```